File: Janken.php
<?php

$ansList = array("グー", "チョキ", "パー");

echo "最初はグー、じゃんけん・・・\n";
echo "あなた：";
$You = trim(fgets(STDIN));
$PHP = $ansList[array_rand($ansList)];
echo "あいて：${PHP}\n";

if ($You == $PHP) echo ("あいこです");
else if ($You == "グー" && $PHP == "チョキ") echo ("あなたの勝ちです");
else if ($You == "グー" && $PHP == "パー") echo ("あなたの負けです");
else if ($You == "チョキ" && $PHP == "グー") echo ("あなたの負けです");
else if ($You == "チョキ" && $PHP == "パー") echo ("あなたの勝ちです");
else if ($You == "パー" && $PHP == "グー") echo ("あなたの勝ちです");
else if ($You == "パー" && $PHP == "チョキ") echo ("あなたの負けです");
else echo ("正しいポーズを選んでください");

echo "\n";

fgets(STDIN);
?>

File: Janken.dart
import 'dart:io';
import 'dart:convert';
import 'dart:math';

void main() async {
  final ansList = ["グー", "チョキ", "パー"];

	print('最初はグー、じゃんけん・・・');
  stdout.write("あなた：");
  final You = stdin.readLineSync(encoding: utf8);
  final m = new Random();
  final numb = m.nextInt(ansList.length);
  final Dart = ansList[numb];
  print("あいて：" + Dart);
  
  if (You == Dart) {
    print("あいこです");
  } else if (You == "グー" && Dart == "チョキ") {
    print("あなたの勝ちです");
  } else if (You == "グー" && Dart == "パー") {
    print("あなたの負けです");
  } else if (You == "チョキ" && Dart == "グー") {
    print("あなたの負けです");
  } else if (You == "チョキ" && Dart == "パー") {
    print("あなたの勝ちです");
  } else if (You == "パー" && Dart == "グー") {
    print("あなたの勝ちです");
  } else if (You == "パー" && Dart == "チョキ") {
    print("あなたの負けです");
  } else {
    print("正しいポーズを選んでください");
  }
}
